src/components/Form/FormFormats.js

```diff
@@ -1648,16 +1701,47 @@ export default {
                         const created = data.createCourse.created;
 
                         if (created) {
-                            const cachedCourses = cache.readQuery({
+                            // Update cache for Registration Courses
+                            const cachedRegistrationCourses = cache.readQuery({
                                 query: GET_ALL_COURSES,
                             });
 
-                            cache.writeQuery({
-                                data: {
-                                    courses: [...cachedCourses, newCourse],
-                                },
-                                query: GET_ALL_COURSES,
-                            });
+                            if (cachedRegistrationCourses !== null) {
+                                cache.writeQuery({
+                                    data: {
+                                        courses: [
+                                            ...cachedRegistrationCourses.courses,
+                                            newCourse,
+                                        ],
+                                    },
+                                    query: GET_ALL_COURSES,
+                                });
+                            }
+
+                            // Update cache for Course Management Courses
+                            const cachedCourseManagementCourses = cache.readQuery(
+                                {
+                                    query: GET_COURSES_BY_ACCOUNT_ID,
+                                    variables: {
+                                        accountId: '',
+                                    },
+                                }
+                            );
+
+                            if (cachedCourseManagementCourses !== null) {
```

Comment:
oh wow. i thought it would usually break on cache.readQuery. this works though?

Reply:
Yeah this works, if that query does not exist in the cache it returns null, so this handles that case.